### PR TITLE
Split Web into API/Dashboard, ping, metric and Rest Provider

### PR DESCRIPTION
**Pull request by Juliens**
This PR changes the Web Provider implementation.

Now we can find all this "internal" routes separately
 - The API / Dashboard
 - The Ping
 - The Metrics
 - Rest Provider

Moreover, all this part, are configured on a entry point.
By doing it like this, we can use auth, whitelister etc.. for all internal routes.
We can attach those internal routes to the same entry point as the reverse proxy entry point

### CODE REVIEW

File: docs/configuration/api.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+|-----------------------------------------------------------------|:-------------:|----------------------------------------------------------------------------------------------------|
+| `/`                                                             |     `GET`     | Provides a simple HTML frontend of Træfik                                                          |
+| `/health`                                                       |     `GET`     | json health metrics                                                                                |
+| `/api`                                                          |     `GET`     | Configuration for all providers                                                                    |
+| `/api/providers`                                                |     `GET`     | Providers                                                                                          |
+| `/api/providers/{provider}`                                     |     `GET`     | Get or update provider                                                                             |
```
Comment (nmengin) [inline docs/configuration/api.md]: Should not we use `PUT` method to update provider?

File: docs/configuration/api.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+| `/api/providers/{provider}/backends/{backend}/servers/{server}` |     `GET`     | Get a server in a backend                                                                          |
+| `/api/providers/{provider}/frontends`                           |     `GET`     | List frontends                                                                                     |
+| `/api/providers/{provider}/frontends/{frontend}`                |     `GET`     | Get a frontend                                                                                     |
+| `/api/providers/{provider}/frontends/{frontend}/routes`         |     `GET`     | List routes in a frontend                                                                          |
+| `/api/providers/{provider}/frontends/{frontend}/routes/{route}` |     `GET`     | Get a route in a frontend                                                                          |
+
+!!!warning
+    For compatibility reason, when you activate the rest provider, you can use `web` or `rest` for `provider` value.
+    But be careful, in the configuration for all provider the key is still `web`
```
Comment (nmengin) [inline docs/configuration/api.md]: all provider**s**

File: docs/configuration/api.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+| `/api/providers/{provider}/backends/{backend}/servers/{server}` |     `GET`     | Get a server in a backend                                                                          |
+| `/api/providers/{provider}/frontends`                           |     `GET`     | List frontends                                                                                     |
+| `/api/providers/{provider}/frontends/{frontend}`                |     `GET`     | Get a frontend                                                                                     |
+| `/api/providers/{provider}/frontends/{frontend}/routes`         |     `GET`     | List routes in a frontend                                                                          |
+| `/api/providers/{provider}/frontends/{frontend}/routes/{route}` |     `GET`     | Get a route in a frontend                                                                          |
+
+!!!warning
+    For compatibility reason, when you activate the rest provider, you can use `web` or `rest` for `provider` value.
```
Comment (nmengin) [inline docs/configuration/api.md]: for `provider` value => **as** `provider` value ?

File: docs/configuration/backends/rest.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## API
+
+| Path                         |     Method    | Description                                    |
+|------------------------------|:-------------:|------------------------------------------------|
+| `/api/providers/web`         |         `PUT` | update provider                                |
+| `/api/providers/rest`        |         `PUT` | update provider                                |
+
+!!!warning
+    For compatibility reason, when you activate the rest provider, you can use `web` or `rest` for `provider` value.
```
Comment (nmengin) [inline docs/configuration/backends/rest.md]: for `provider` value => **as** `provider` value ?

File: docs/configuration/ping.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+
+| Path                                                            |     Method    | Description                                                                                        |
+|-----------------------------------------------------------------|:-------------:|----------------------------------------------------------------------------------------------------|
+| `/ping`                                                         | `GET`, `HEAD` | A simple endpoint to check for Træfik process liveness. Return a code `200` with the content: `OK` |
+
+
+!!!warning
+    Even if you have authentication configure on entrypoint, the `/ping` path of the api is excluded from authentication.
```
Comment (nmengin) [inline docs/configuration/ping.md]: configure**d**

File: integration/basic_test.go [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
-	"syscall"
 	"time"
 
+	"fmt"
 	"github.com/containous/traefik/integration/try"
 	"github.com/go-check/check"
 	checker "github.com/vdemeester/shakers"
+	"os"
+	"strings"
+	"syscall"
```
Comment (nmengin) [inline integration/basic_test.go]: Sort imports :wink:

File: version/version.go [hunk 7]
```diff
@@ -4,9 +4,13 @@ import (
 	"context"
 	"net/url"
 
+	"net/http"
```
Comment (nmengin) [inline version/version.go]: Sort imports :wink:

File: provider/rest/rest.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/containous/mux"
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	"github.com/unrolled/render"
+)
+
+// Provider is a provider.Provider implementation that provides the UI
+type Provider struct {
```
Comment (nmengin) [inline provider/rest/rest.go]: We should change the comment because the provider does not provides `UI` yet, isn't it?

**Comment by shadycuz** [comment]
Does this mean I can leave my /metrics exposed for Prometheus while still providing auth for dashboard? I was going to open a feature request for that.

**Comment by Juliens** [comment]
@shadycuz yes but on different entrypoints (not the same port)

**Comment by tcoupin** [comment]
No more path option like in #1233 ?

**Comment by Juliens** [comment]
@tcoupin the path option disappear but you can do it like this http://v1-5.archive.docs.traefik.io/configuration/backends/web/#path

**Pull request merged**